core/trino-main/src/main/java/io/trino/operator/Driver.java

```diff
@@ -727,16 +727,40 @@ private <T> Optional<T> tryWithLock(long timeout, TimeUnit unit, boolean interru
                 try {
                     processNewSources();
                 }
-                finally {
+                catch (Throwable t) {
+                    if (failure == null) {
+                        failure = t;
+                    }
+                    else if (failure != t) {
+                        failure.addSuppressed(t);
+                    }
+                }
+
+                try {
                     destroyIfNecessary();
                 }
+                catch (Throwable t) {
+                    if (failure == null) {
+                        failure = t;
+                    }
+                    else if (failure != t) {
+                        failure.addSuppressed(t);
+                    }
+                }
             }
             finally {
                 exclusiveLock.unlock();
             }
         }
 
-        return result;
+        if (failure != null) {
+            throwIfUnchecked(failure);
+            // should never happen
+            throw new RuntimeException(failure);
+        }
+
+        verify(result != null, "result is null");
```

Comment:
Not necessary. The `Optional.of()` construction will fail if it's null.

Reply:
I added this more as a documentation so the intention is clearly stated. Otherwise when some task starts returning `null` somebody may interpret the `Optional.of` as a mistake and switch it to `Optional.ofNullable` instead of fixing the actual problem.